state-sync/state-sync-v2/state-sync-driver/src/bootstrapper.rs

```diff
@@ -947,7 +968,12 @@ impl<
         // Initialize the state value synchronizer (if not already done)
         if !self.state_value_syncer.initialized_state_snapshot_receiver {
             // Fetch all verified epoch change proofs
-            let epoch_change_proofs = self.verified_epoch_states.all_epoch_ending_ledger_infos();
+            let version_to_sync = ledger_info_to_sync.ledger_info().version();
+            let epoch_change_proofs = if version_to_sync == GENESIS_TRANSACTION_VERSION {
+                vec![ledger_info_to_sync.clone()] // Sync to genesis
+            } else {
+                self.verified_epoch_states.all_epoch_ending_ledger_infos() // Sync beyond genesis
```

Comment:
curious why introducing a branch here. what is contained in the all_epoch_ending_ledger_infos when the version_to_sync is genesis?

Reply:
Aah, it's because `verified_epoch_states.all_epoch_ending_ledger_infos` only contains epoch ending ledger infos beyond genesis. Thus, when the version to sync is genesis, `verified_epoch_states.all_epoch_ending_ledger_infos` is empty.